src/main/webapp/feed.html

```diff
@@ -23,24 +23,48 @@ <h1>Message Feed</h1>
 
     function MessageFeed() {
       const [allMessages, setMessages] = React.useState([])
+      const [numMessagesPerUser, setNumUserMessages] = React.useState(["No messages!", h('br')])
 
       React.useEffect(fetchMessages, []);
 
       // Fetch messages and add them to the page.
       function fetchMessages() {
+        // store message react components
         const fetchedMessages = []
+        // store how many messages each user posts in map for later use
+        const userMessageCounts = {}
+        // store react components of the num of msgs each user posts
+        const finalUserMessageCounts = ["Number of Messages Per User:", h('br')]
         const url = '/feed';
         fetch(url)
           .then(response => response.json())
           .then(messages => {
+            // get data from each message
             messages.forEach(message => {
+              // if a user has posted before, incr
```

Comment:
Please avoid abbreviations in comments unless they are extremely clear and not abbreviating would be overly verbose (e.g. it's okay to abbreviate "HTML").